pkg/agent/openflow/network_policy.go

```diff
@@ -2220,3 +2228,69 @@ func (f *featureNetworkPolicy) l7NPTrafficControlFlows() []binding.Flow {
 			Done(),
 	}
 }
+
+func (f *featureNetworkPolicy) loggingNPPacketFlowWithOperations(cookieID uint64, operations uint8) binding.Flow {
+	loggingOperations := binding.NewRegMark(PacketInOperationField, uint32(operations))
+	fb := L2ForwardingOutTable.ofTable.BuildFlow(priorityNormal).
+		MatchRegMark(SendToControllerRegMark, loggingOperations)
+	if f.ovsMetersAreSupported {
+		fb = fb.Action().Meter(PacketInMeterIDNP)
+	}
+	return fb.Action().SendToController([]byte{uint8(PacketInCategoryNP), operations}, false).
+		Cookie(cookieID).
+		Done()
+}
+
+func (f *featureNetworkPolicy) initLoggingFlows() []binding.Flow {
+	maxOperationValue := PacketInNPLoggingOperation + PacketInNPStoreDenyOperation + PacketInNPRejectOperation
+	flows := make([]binding.Flow, 0, maxOperationValue)
+	if !f.enableAntreaPolicy {
+		return flows
+	}
```

Comment:
The check should be in the begining, it can just return nil.

Reply:
updated.